### PR TITLE
Summoner Improvements: Odin, Zantetsuken, and Atomos

### PR DESCRIPTION
**Pull request by genoxd**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

I have fixed Odin/Zantetsuken for Summoner and adjusted the timings for Atomos. 
* Odin now correctly spawns, performs Zantetsuken, and despawns.
* Odin no longer hits mobs with melee hits.
* The death formula for Zantetsuken was incorrect. I have fixed it: KO chance  = (MP/MaxMP)/NumberOfTargets.
  * [Wiki Source](https://ffxiclopedia.fandom.com/wiki/Odin)
  * Also I've been SMN since the dawn of time and have used Odin excessively for solo Job Points and various content where he doesn't suck like Incursion.
* I have updated the timing for Atomos so that it better matches retail.

## Steps to test these changes

* Switch to SMN level 99 with Odin
* Have full MP
* Use Astral Flow
* Cast Odin on a single mob and confirm death rate is 100%
* Cast Odin on multiple mobs and confirm death rate is not 100% (I repeated this to confirm that it can miss all mobs and hit some mobs)

* Have low MP
* Use Astral Flow
* Cast Odin on a single mob and confirm death rate is NOT 100%

Edit: Forgot to mention I got the animation id  from retail and I lined up a 5 mobs in a row then cast odin in a line to guesstimate the range. It's between 14-17 yalms based on which mobs got hit in my test, so I am guessing it's actually 15 yalms.

### CODE REVIEW

**Comment by Omnione** [comment]
out of curiosity, have you tested Atomos being out of range or the target being dead before Deconstruction fires off?, from testing when i added this, if i had the Chronoshift ability fire off in the Deconstruction script he would not be able to cast Deconstruction so Chronoshift would not go off and Atomos would just stand there doing nothing as the despawn code never gets fired

**Comment by genoxd** [comment]
> out of curiosity, have you tested Atomos being out of range or the target being dead before Deconstruction fires off?, from testing when i added this, if i had the Chronoshift ability fire off in the Deconstruction script he would not be able to cast Deconstruction so Chronoshift would not go off and Atomos would just stand there doing nothing as the despawn code never gets fired

Good catch! I didn't and Odin might have the same issue if a mob runs out of range. I'll test both today and fix if broken

**Comment by genoxd** [comment]
I did more testing around what happens if the target is out of range and if the summoner is out of range. I've updated this to reflect what I saw on retail.

Also fixed the issue where if the ability fails to fire the pet could end up in a bad state (never depsawning).

Odin: Odin always despawns 15 seconds after being summoned even if he doesn't perform Zantetsuken.

Atomos: Atomos always despawns ~17 seconds after being summoned. Deconstruction is used if the target is in range. Chronoshift is used even if Deconstruction isn't used.


Edit: I also tested killing the mob before Atomos does Deconstruction and it still uses Chronoshift, same behavior as being out of range

File: scripts/actions/abilities/pets/zantetsuken.lua
Comment (Skold177) [inline scripts/actions/abilities/pets/zantetsuken.lua:1]: "Damage to NMs caps at 9999 or 10-20% of their current HP, whichever is lower. However, percentage damage multipliers are applied after this calculation. This is why damage may exceed 9999."

Unless you checked it on retail, be sure to clamp the damage to 10% of the targets HP vs NMs.

Comment (genoxd) [reply scripts/actions/abilities/pets/zantetsuken.lua:1]: Sure, I didn't modify the damage portion but I can attempt to fix that too

Comment (sruon) [inline scripts/actions/abilities/pets/zantetsuken.lua:60]: JA_MISS_2                       = 324, -- <user> uses <ability>, but misses <target>. (includes target name)

Image

File: sql/pet_skills.sql [hunk 1]
```diff
@@ -187,7 +187,7 @@ INSERT INTO `pet_skills` VALUES (666,0,155,'pavor_nocturnus',0,0,12,2000,1000,4,
 INSERT INTO `pet_skills` VALUES (667,0,173,'blindside',0,0,10,2000,1000,4,317,@SKILLFLAG_BLOODPACT_RAGE,0,13,0,@SC_GRAVITATION,@SC_TRANSFIXION,0); -- No LUA yet
 INSERT INTO `pet_skills` VALUES (668,0,171,'deconstruction',0,0,21,2000,1000,4,323,@SKILLFLAG_BLOODPACT_RAGE,0,13,0,0,0,0);
 INSERT INTO `pet_skills` VALUES (669,0,172,'chronoshift',1,14,0,2000,1000,3,283,@SKILLFLAG_BLOODPACT_WARD,0,13,0,0,0,0);
--- 670: Zantetsuken 10y (Radius 10y)
+INSERT INTO `pet_skills` VALUES (670,0,151,'zantetsuken',4,15,15.0,2000,1000,4,277,@SKILLFLAG_ASTRAL_FLOW | @SKILLFLAG_BLOODPACT_RAGE,0,13,0,0,0,0); -- Conal 15y, anim/msg from capture
```
Comment (sruon) [inline sql/pet_skills.sql:190]: Do you have a capture proving this is 15y on both range and radius?

I can only prove 10y so far on both range and radius (the same values the DATs claim). Given it's a special type of move I'm willing to believe they may not be right.

Furthermore 15y isn't used by any other ability in the game (closest values would be 14 or 16).

Comment (genoxd) [reply sql/pet_skills.sql:190]: I stated it in the PR: "Forgot to mention I got the animation id from retail and I lined up a 5 mobs in a row then cast odin in a line to guesstimate the range. It's between 14-17 yalms based on which mobs got hit in my test, so I am guessing it's actually 15 yalms." 

If all other abilities are 14 or 16, then I don't have enough evidence to prove it. It killed something at 14, it did not kill something at just over 17. It could be 16. I can check later. Radius doesn't make sense since this is a conal attack

File: scripts/actions/abilities/pets/zantetsuken.lua
Comment (genoxd) [reply scripts/actions/abilities/pets/zantetsuken.lua:60]: Unsure what you're asking for here

Comment (genoxd) [reply scripts/actions/abilities/pets/zantetsuken.lua:1]: Follow up question: Are there any relatively easy NMs to access on retail that we have accurate stats for? If I'm going to redo the damage equation, I'd like to have a baseline to work with ideally
